server/src/main/java/org/opensearch/search/query_group/QueryGroupServiceSettings.java

```diff
@@ -0,0 +1,256 @@
+     */
+    public static final String NODE_MEMORY_CANCELLATION_THRESHOLD_SETTING_NAME = "query_group.node.memory_cancellation_threshold";
+    /**
+     * Setting to control the memory based cancellation threshold
+     */
+    public static final Setting<Double> NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD = Setting.doubleSetting(
+        NODE_MEMORY_CANCELLATION_THRESHOLD_SETTING_NAME,
+        DEFAULT_NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    /**
+     * Setting name for node level cpu based cancellation threshold for QueryGroup service
+     */
+    public static final String NODE_CPU_CANCELLATION_THRESHOLD_SETTING_NAME = "query_group.node.cpu_cancellation_threshold";
+    /**
+     * Setting to control the cpu based cancellation threshold
+     */
+    public static final Setting<Double> NODE_LEVEL_CPU_CANCELLATION_THRESHOLD = Setting.doubleSetting(
+        NODE_CPU_CANCELLATION_THRESHOLD_SETTING_NAME,
+        DEFAULT_NODE_LEVEL_CPU_CANCELLATION_THRESHOLD,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    /**
+     * QueryGroup service settings constructor
+     * @param settings - QueryGroup service settings
+     * @param clusterSettings - QueryGroup cluster settings
+     */
+    public QueryGroupServiceSettings(Settings settings, ClusterSettings clusterSettings) {
+        nodeLevelMemoryCancellationThreshold = NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD.get(settings);
+        nodeLevelMemoryRejectionThreshold = NODE_LEVEL_MEMORY_REJECTION_THRESHOLD.get(settings);
+        nodeLevelCpuCancellationThreshold = NODE_LEVEL_CPU_CANCELLATION_THRESHOLD.get(settings);
+        nodeLevelCpuRejectionThreshold = NODE_LEVEL_CPU_REJECTION_THRESHOLD.get(settings);
+
+        ensureRejectionThresholdIsLessThanCancellation(
+            nodeLevelMemoryRejectionThreshold,
+            nodeLevelMemoryCancellationThreshold,
+            NODE_MEMORY_REJECTION_THRESHOLD_SETTING_NAME,
+            NODE_MEMORY_CANCELLATION_THRESHOLD_SETTING_NAME
+        );
+        ensureRejectionThresholdIsLessThanCancellation(
+            nodeLevelCpuRejectionThreshold,
+            nodeLevelCpuCancellationThreshold,
+            NODE_CPU_REJECTION_THRESHOLD_SETTING_NAME,
+            NODE_CPU_CANCELLATION_THRESHOLD_SETTING_NAME
+        );
+
+        clusterSettings.addSettingsUpdateConsumer(NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD, this::setNodeLevelMemoryCancellationThreshold);
+        clusterSettings.addSettingsUpdateConsumer(NODE_LEVEL_MEMORY_REJECTION_THRESHOLD, this::setNodeLevelMemoryRejectionThreshold);
+        clusterSettings.addSettingsUpdateConsumer(NODE_LEVEL_CPU_CANCELLATION_THRESHOLD, this::setNodeLevelCpuCancellationThreshold);
+        clusterSettings.addSettingsUpdateConsumer(NODE_LEVEL_CPU_REJECTION_THRESHOLD, this::setNodeLevelCpuRejectionThreshold);
```

Comment:
It generally makes sense for the Settings to sit in the same class consuming these settings. Else, we need to have the settings update consumer at multiple places or expose the settings getter elsewhere

Reply:
I think it should be fine for the settings to sit in a single place. Why can't those setters sit in the dedicated class itself. and whoever wants to use these settings can consume this .

Reply:
Fair enough, we can have use the public getter via the instance injected in other classes and have the settings update consumer in this class!